src/mint/process.rs

```diff
@@ -135,35 +136,73 @@ pub fn process_mint(args: MintArgs) -> Result<()> {
     } else {
         let pb = progress_bar_with_style(number);
 
+        let mut tasks = Vec::new();
+        let semaphore = Arc::new(Semaphore::new(100));
+        let config = Arc::new(sugar_config);
+
         for _i in 0..number {
-            if let Err(err) = mint(
-                Arc::clone(&client),
-                candy_pubkey,
-                Arc::clone(&candy_machine_state),
-                Arc::clone(&collection_pda_info),
-                receiver_pubkey,
-            ) {
-                pb.abandon_with_message(format!("{}", style("Mint failed ").red().bold()));
-                error!("{:?}", err);
-                return Err(err);
-            }
+            let config = config.clone();
+            let permit = Arc::clone(&semaphore).acquire_owned().await.unwrap();
+            let candy_machine_state = candy_machine_state.clone();
+            let collection_pda_info = collection_pda_info.clone();
+            let pb = pb.clone();
+
+            // Start tasks
+            tasks.push(tokio::spawn(async move {
+                let _permit = permit;
+                let res = mint(
+                    config,
+                    candy_pubkey,
+                    candy_machine_state,
+                    collection_pda_info,
+                    receiver_pubkey,
+                )
+                .await;
+                pb.inc(1);
+                res
+            }));
+        }
+
+        let mut error_count = 0;
 
-            pb.inc(1);
+        // Resolve tasks
+        for task in tasks {
+            let res = task.await.unwrap();
+            if let Err(e) = res {
+                error_count += 1;
+                error!("{:?}, continuing. . .", e);
+            }
         }
 
-        pb.finish();
+        if error_count > 0 {
+            pb.abandon_with_message(format!(
+                "{} {} items failed.",
+                style("Some of the items failed to mint.").red().bold(),
+                error_count
+            ));
+            return Err(anyhow!(
+                "{} {}/{} {}",
+                style("Minted").red().bold(),
+                number - error_count,
+                number,
+                style("of the items").red().bold()
+            ));
+        }
     }
 
+    pb.finish();
```

Comment:
This line needs to move up between the `}` and `}` from lines 190-191, since there is already a `pb.finish*` statement for the first branch of the `if` statement on line 135. Having this here creates an extra line on the output since it ends up referencing the previous `pb` object from line 75 (which is already finished).

Reply:
Good catch!